### PR TITLE
[highs] Enable shared build on Windows

### PR DESCRIPTION
**Pull request by hedtke**
Specify library name and version:  **highs**

Enable shared build on Windows and drop v1 test package

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/highs/all/conanfile.py [hunk 1]
```diff
@@ -1,14 +1,11 @@
 from conan import ConanFile
 from conan.tools.apple import fix_apple_shared_install_name
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
-from conan.errors import ConanInvalidConfiguration
 from conan.tools.files import apply_conandata_patches, collect_libs, copy, export_conandata_patches, get
 from conan.tools.microsoft import is_msvc
 from conan.tools.scm import Version
 from os.path import join
 
-required_conan_version = ">=1.53.0"
```
Comment (RubenRBS) [inline recipes/highs/all/conanfile.py:10]: We want to keep this, it provides a nicer error for old versions :)

File: recipes/highs/all/conanfile.py [hunk 2]
```diff
@@ -27,10 +24,6 @@ class HiGHSConan(ConanFile):
         "fPIC": True,
     }
 
-    def validate(self):
```
Comment (RubenRBS) [inline recipes/highs/all/conanfile.py:30]: So I take that there was no issue with it and ir always worked? Oh

**Review by RubenRBS — APPROVED** [review]
Weird that it works, I wonder why it was disabled in the first case then 🤔

File: recipes/highs/all/conanfile.py (hunk 1 above)
Comment (hedtke) [reply recipes/highs/all/conanfile.py]: Added again

File: recipes/highs/all/conanfile.py (hunk 2 above)
Comment (hedtke) [reply recipes/highs/all/conanfile.py:30]: I forgot to copy the .lib file on Windows in the initial version of the recipe. I didn't know how to fix the problem of missing symbols. My only solution was to deactivate shared libs on Windows. I know better now.

**Review by RubenRBS — APPROVED** [review]
Thanks! Sorry for the delay, I missed the notification about the pending review after the new commit :)

**Review by danimtb — APPROVED** [review]

**Pull request merged**